buildSrc/src/main/java/com/google/firebase/gradle/plugins/DackkaPlugin.md

```diff
@@ -0,0 +1,99 @@
+# Dackka Plugin
+
+If the implementation of the DackkaPlugin is not straight forward enough
+for you to understand, hopefully this document will help you.
+
+## Dackka
+
+Dackka is an internal-purposed Dokka plugin. Google hosts its documentation on
+an internal service called **Devsite**. Firebase hosts their documentation on a
+variant of Devsite called **Firesite**. You can click [here](https://firebase.google.com/docs/reference) 
+to see how that looks. Essentially, it's just Google's way of decorating (and organizing)
+documentation.
+
+Devsite expects its files to be in a very specific format. Previously, we would
+use an internal variant of metalava called doclava- which allowed us to provide
+sensible defaults as to how the Javadoc should be rendered. Then, we would do
+some further transformations to get the Javadoc output in-line with what
+Devsite expects. This was a lengthy process, and came with a lot of overhead.
```

Comment:
Actually doclava and metalava are unrelated, [doclava](https://code.google.com/archive/p/doclava/) is a javadoc doclet that has been abandoned and is currently unmaintained, metalava is a reimplementation of a subset of what doclava's functionality that is unrelated to javadoc generation but rather api surface management.

So I would suggest changing this to something along the lines of
```suggestion
Devsite expects its files to be in a very specific format. Previously, we would
use a custom javadoc doclet doclava- which allowed us to provide
sensible defaults as to how the Javadoc should be rendered. Then, we would do
some further transformations to get the Javadoc output in-line with what
Devsite expects. However doclava does not support kotlindoc and has been unmaintained for many years.
```

Reply:
Hmm, im a bit confused then. Would you mind explaining that some more? What purpose does metalava provide in our doc generation? My understanding is it handles removing classes marked with hidden and provides stub variants of the source?

Reply:
yeah that's right, that's what metalava does for us, additionally it produces api.txt files for us, this is what doclava does as part of javadoc generation as well.

Reply:
Where do we use api.txt files again? Also, do we have any purpose for the stubs beyond javadoc generation?